rhp/v4/siamux/siamux.go

```diff
@@ -56,16 +56,20 @@ func Dial(ctx context.Context, addr string, peerKey types.PublicKey) (rhp4.Trans
 	if err != nil {
 		return nil, fmt.Errorf("failed to connect to %q: %w", addr, err)
 	}
-	if deadline, ok := ctx.Deadline(); ok {
-		if err := conn.SetDeadline(deadline); err != nil {
-			return nil, fmt.Errorf("failed to set deadline: %w", err)
-		}
-	}
```

Comment:
This was actually counter-productive. Setting a deadline on a stream is fine since it serves only a single purpose but the connection potentially lives for a lot longer. So just because we interrupt the dial after 30s, 60s, or even longer, we shouldn't be killing the mux with a deadline after that time.